### PR TITLE
feat(headless, folding): Adapted the headless folded result list controller to support insight use

### PR DESCRIPTION
**Pull request by SimonMilord**
[SFINT-4851](https://coveord.atlassian.net/browse/SFINT-4851)

**As a User**
I want to use the folding feature when I’m using Quantic for the insight use case. I need to be able to build a Headless folded result list controller using an insight engine.

**Acceptance criteria**
Headless folded result list controller should work with the insight use case.

**Technical details**
We need to create a core headless folded result list controller headless that accepts both a search engine and an insight engine. This core controller would then be used to build a folded result list controller for both the search use case and the insight case.

**STEPS** (maybe will make it easier to understand what is going on in the PR): 

1. Move analytics actions in their own file
2. Added analytics actions for insight folding
3. Updated exports files (index.ts and insight.index.ts) for folding
4. Added Core controller
5. Added InsightFoldedResultList controller
6. Refactored existing controller to use SearchEngine. Cleaned up file
7. Added Insight-folding-actions and its loader
8. wrote unit tests for the core controller
9. wrote unit tests for insight analytics

**DEMO (search & insight usecases):**

https://github.com/coveo/ui-kit/assets/73316533/f7c00372-8143-4802-8cc3-93b0cd00c999

[SFINT-4851]: https://coveord.atlassian.net/browse/SFINT-4851?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/headless/src/controllers/core/folded-result-list/headless-core-folded-result-list.test.ts [hunk 1]
```diff
@@ -0,0 +1,145 @@
+import {search, configuration, folding, query} from '../../../app/reducers';
```
Comment (olamothe) [inline packages/headless/src/controllers/core/folded-result-list/headless-core-folded-result-list.test.ts:1]: You'll see the `reducers` file has been deleted in master as of today.

This is part of an initiative to reduce bundle size for our clients. You will need to imports reducers from their respective files instead 

ie: 
```typescript
import {searchReducer as search} from '../../../features/search/search-slice'`
// etc.
```

File: package.json [hunk 2]
```diff
@@ -22,6 +22,7 @@
     "notify:docs": "node ./scripts/notify-docs/published-ui-kit.mjs"
   },
   "dependencies": {
+    "coveo.analytics": "^2.27.12",
```
Comment (mmitiche) [inline package.json:25]: This should not be here.
`npm i coveo.analytics@xyz` should be executed from ui-kit/packages/headless and the dependency  should be updated in `ui-kit/packages/quantic/package.json`

File: packages/headless/src/controllers/core/folded-result-list/headless-core-folded-result-list.ts [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+  moreResultsAvailable: boolean;
+}
+
+export interface FoldedResultAnalyticsClient {
+  logShowMoreFoldedResults: (result: Result) => ClickAction;
+  logShowLessFoldedResults: () => CustomAction;
+}
+
+/**
+ * Creates a `FoldedResultList` controller instance.
```
Comment (mmitiche) [inline packages/headless/src/controllers/core/folded-result-list/headless-core-folded-result-list.ts:158]:
```suggestion
 * Creates a core `FoldedResultList` controller instance.
```

File: packages/headless/src/controllers/folded-result-list/headless-folded-result-list.ts [hunk 4]
```diff
@@ -148,111 +42,18 @@ export interface FoldedResultListState extends SearchStatusState {
  */
 export function buildFoldedResultList(
   engine: SearchEngine,
-  props: FoldedResultListProps = {}
+  props: FoldedResultListProps
```
Comment (mmitiche) [inline packages/headless/src/controllers/folded-result-list/headless-folded-result-list.ts:45]: why do we need to remove this default value?

File: packages/headless/src/features/insight-folding/insight-folding-actions-loader.ts
Comment (mmitiche) [inline packages/headless/src/features/insight-folding/insight-folding-actions-loader.ts:1]: I don't think the folder`features/insight-folding` should be created.
You can directly add the file `folding-insight-actions-loader.ts` inside the folder `features/folding`.

File: packages/headless/src/features/insight-search/insight-search-request.ts [hunk 5]
```diff
@@ -55,6 +56,10 @@ export const buildInsightSearchRequest = (
       sortCriteria: state.sortCriteria,
     }),
     tab: state.configuration.analytics.originLevel2,
+    filterField: state.folding?.fields.collection,
+    childField: state.folding?.fields.parent,
+    parentField: state.folding?.fields.child,
+    filterFieldRange: 100,
```
Comment (mmitiche) [inline packages/headless/src/features/insight-search/insight-search-request.ts:62]: This should not always be 100. The value of this parameter should be 100 only in the `loadCollection` action.
```suggestion
    filterFieldRange: state.folding.filterFieldRange,
```

**Review by mmitiche — COMMENTED** [review]
taking shape 👌
A bit more work needed for the load collection action and the unit test but nice job!

File: package.json (hunk 2 above)
Comment (SimonMilord) [reply package.json:25]: Yeah I forgot to cleanup this, will remove

File: packages/headless/src/controllers/folded-result-list/headless-folded-result-list.ts (hunk 4 above)
Comment (mmitiche) [reply packages/headless/src/controllers/folded-result-list/headless-folded-result-list.ts:45]: I think this is a breaking change, we shall not touch the signature of the `buildFoldedResultList`.

File: packages/headless/src/controllers/core/folded-result-list/headless-core-folded-result-list.ts [hunk 6]
```diff
@@ -0,0 +1,275 @@
...
+   * @defaultValue `2`
+   */
+  numberOfFoldedResults?: number;
+}
+
+export interface FoldedResultListOptions extends ResultListOptions {
+  folding?: FoldingOptions;
+}
+
+export interface FoldedResultListProps {
```
Comment (mmitiche) [inline packages/headless/src/controllers/core/folded-result-list/headless-core-folded-result-list.ts:71]: You  need to name this `CoreFoldedResultListProps` because that's not the interface that we need to pass to the search/insight folding result controller.
The interface we need to pass to the search/insight folding result controller doesn’t need to contain `loadCollectionActionCreator`, it will just be: 
```
export interface FoldedResultListProps {
  /**
   * The options for the `FoldedResultList` controller.
   * */
  options?: FoldedResultListOptions;
}
```

File: packages/headless/src/controllers/core/folded-result-list/headless-core-folded-result-list.ts [hunk 7]
```diff
@@ -0,0 +1,275 @@
...
+  folding?: FoldingOptions;
+}
+
+export interface CoreFoldedResultListProps {
+  /**
+   * The options for the `FoldedResultList` controller.
+   * */
+  options?: FoldedResultListOptions;
+  /**
+   *
```
Comment (SimonMilord) [inline packages/headless/src/controllers/core/folded-result-list/headless-core-folded-result-list.ts:77]: just noticed missing description here, will add

File: packages/headless/src/api/search/search-metadata.ts [hunk 8]
```diff
@@ -6,7 +6,8 @@ export type SearchApiMethod =
   | 'recommendations'
   | 'html'
   | 'productRecommendations'
-  | 'fieldDescriptions';
+  | 'fieldDescriptions'
+  | 'query';
```
Comment (mmitiche) [inline packages/headless/src/api/search/search-metadata.ts:10]: I don't  think the new value here is necessary, it's not a new Search API method that you are using for folding, you are doing a search query so I think you can just use the value `search` in you request.

File: packages/headless/src/api/service/insight/insight-api-client.ts [hunk 9]
```diff
@@ -81,10 +82,12 @@ export class InsightAPIClient {
   }
 
   async query(
-    req: InsightQueryRequest
+    req: InsightQueryRequest,
+    options?: SearchOptions
   ): Promise<InsightAPIResponse<SearchResponseSuccess>> {
     const response = await PlatformClient.call({
       ...buildInsightQueryRequest(req),
+      requestMetadata: {method: 'query', origin: options?.origin},
```
Comment (mmitiche) [inline packages/headless/src/api/service/insight/insight-api-client.ts:90]:
```suggestion
      requestMetadata: {method: 'search', origin: options?.origin},
```

File: packages/headless/src/controllers/result-list/headless-result-list.ts [hunk 10]
```diff
@@ -18,7 +18,7 @@ export type {ResultListOptions, ResultListProps, ResultListState, ResultList};
  * @returns A `ResultList` controller instance.
  */
 export function buildResultList(
-  engine: SearchEngine,
+  engine: CoreEngine,
```
Comment (mmitiche) [inline packages/headless/src/controllers/result-list/headless-result-list.ts:21]: This is the function that builds the Headless Search Result controller, I don't think it should receive a CoreEngine. it's specific to search.

File: packages/headless/src/controllers/core/folded-result-list/headless-core-folded-result-list.ts [hunk 11]
```diff
@@ -0,0 +1,275 @@
...
+export function buildCoreFoldedResultList(
+  engine: CoreEngine,
+  props: CoreFoldedResultListProps,
+  analyticsClient: FoldedResultAnalyticsClient
+): CoreFoldedResultList {
+  if (!loadFoldingReducer(engine)) {
+    throw loadReducerError;
+  }
+
+  const controller = buildResultList(engine, props);
```
Comment (mmitiche) [inline packages/headless/src/controllers/core/folded-result-list/headless-core-folded-result-list.ts:174]: Instead of making `buildResultList` use a core engine which is not ideal, you can use the `buildCoreResultList`.

File: packages/headless/src/features/folding/insight-folding-actions.ts [hunk 12]
```diff
@@ -0,0 +1,106 @@
...
+}
+
+export const foldingOptionsSchemaDefinition: SchemaDefinition<
+  Required<RegisterFoldingActionCreatorPayload>
+> = {
+  collectionField: new StringValue({emptyAllowed: false, required: false}),
+  parentField: new StringValue({emptyAllowed: false, required: false}),
+  childField: new StringValue({emptyAllowed: false, required: false}),
+  numberOfFoldedResults: new NumberValue({min: 0, required: false}),
+};
```
Comment (mmitiche) [inline packages/headless/src/features/folding/insight-folding-actions.ts:58]: You can import these interfaces from `folding-actions.ts` instead.

File: packages/headless/src/features/folding/insight-folding-actions.ts [hunk 13]
```diff
@@ -0,0 +1,106 @@
...
+export const registerFolding = createAction(
+  'folding/register',
+  (payload: RegisterFoldingActionCreatorPayload) =>
+    validatePayload(payload, foldingOptionsSchemaDefinition)
+);
+
+export type StateNeededByLoadCollection = ConfigurationSection &
+  FoldingSection &
+  QuerySection &
+  InsightConfigurationSection;
```
Comment (mmitiche) [inline packages/headless/src/features/folding/insight-folding-actions.ts:69]: Same for this interface, we can import it and save some lines of code.

File: packages/headless/src/features/insight-search/insight-search-request.ts [hunk 14]
```diff
@@ -55,6 +56,10 @@ export const buildInsightSearchRequest = (
       sortCriteria: state.sortCriteria,
     }),
     tab: state.configuration.analytics.originLevel2,
+    filterField: state.folding?.fields.collection,
+    childField: state.folding?.fields.parent,
+    parentField: state.folding?.fields.child,
+    filterFieldRange: state.folding?.filterFieldRange,
```
Comment (mmitiche) [inline packages/headless/src/features/insight-search/insight-search-request.ts:62]:
```suggestion
     ...(state.folding && {
          filterField: state.folding.fields.collection,
         childField: state.folding.fields.parent,
         parentField: state.folding.fields.child,
         filterFieldRange: state.folding.filterFieldRange,
    }),
```

**Review by mmitiche — COMMENTED** [review]
Nice job! few things to address but it's taking shape👌

File: packages/headless/src/controllers/result-list/headless-result-list.ts (hunk 10 above)
Comment (SimonMilord) [reply packages/headless/src/controllers/result-list/headless-result-list.ts:21]: you are right, must have forgotten this one as the insight has a InsightSearchEngine. Good catch

File: packages/headless/src/features/folding/insight-folding-actions.ts (hunk 12 above)
Comment (SimonMilord) [reply packages/headless/src/features/folding/insight-folding-actions.ts:58]: touché :D

File: packages/headless/src/features/folding/insight-folding-actions.ts (hunk 13 above)
Comment (SimonMilord) [reply packages/headless/src/features/folding/insight-folding-actions.ts:69]: This one I put there since we add the InsightConfigurationSection

File: packages/headless/src/controllers/core/folded-result-list/headless-core-folded-result-list.ts [hunk 15]
```diff
@@ -0,0 +1,277 @@
...
+): CoreFoldedResultList {
+  if (!loadFoldingReducer(engine)) {
+    throw loadReducerError;
+  }
+
+  const controller = buildCoreResultList(engine, props);
+  const {dispatch} = engine;
+  const getState = () => engine.state;
+
+  const options = props?.options?.folding
```
Comment (btaillon) [inline packages/headless/src/controllers/core/folded-result-list/headless-core-folded-result-list.ts:180]: Not sure why this optional chain is needed, `props` is required.
```suggestion
  const options = props.options?.folding
```

File: packages/headless/src/controllers/core/folded-result-list/headless-core-folded-result-list.ts [hunk 16]
```diff
@@ -0,0 +1,277 @@
...
+
+  return {
+    ...controller,
+
+    loadCollection: (collection) => {
+      dispatch(
+        props.loadCollectionActionCreator(
+          collection?.result?.raw[
+            engine.state.folding.fields.collection
+          ] as string
```
Comment (btaillon) [inline packages/headless/src/controllers/core/folded-result-list/headless-core-folded-result-list.ts:199]: Same as above. I don't see why this optional chain is needed, since `collection` and `result` both seem required.

```suggestion
          collection.result.raw[
            engine.state.folding.fields.collection
          ] as string
```

File: packages/headless/src/controllers/core/folded-result-list/headless-core-folded-result-list.ts [hunk 17]
```diff
@@ -0,0 +1,277 @@
...
+    },
+  };
+}
+
+function loadFoldingReducer(
+  engine: CoreEngine
+): engine is CoreEngine<
+  SearchSection & ConfigurationSection & FoldingSection & QuerySection
+> {
+  engine.addReducers({search, configuration, folding, query});
```
Comment (btaillon) [inline packages/headless/src/controllers/core/folded-result-list/headless-core-folded-result-list.ts:256]: Are all of these reducers needed for the insight use case?

File: packages/headless/src/features/folding/folding-actions.ts [hunk 18]
```diff
@@ -73,7 +66,8 @@ export const registerFolding = createAction(
 
 export type StateNeededByLoadCollection = ConfigurationSection &
   FoldingSection &
-  QuerySection;
+  QuerySection &
+  InsightConfigurationSection;
```
Comment (btaillon) [inline packages/headless/src/features/folding/folding-actions.ts:70]: Am I correct that this `loadCollection` action should only be used by the search use case, and the insight use case uses a different one?

If so, you should revert this change.
```suggestion
  QuerySection;
```

File: packages/headless/src/features/folding/insight-folding-actions.ts [hunk 19]
```diff
@@ -0,0 +1,68 @@
...
+export const registerFolding = createAction(
+  'folding/register',
+  (payload: RegisterFoldingActionCreatorPayload) =>
+    validatePayload(payload, foldingOptionsSchemaDefinition)
+);
+
+// export type StateNeededByLoadCollection = ConfigurationSection &
+//   FoldingSection &
+//   QuerySection &
+//   InsightConfigurationSection;
```
Comment (btaillon) [inline packages/headless/src/features/folding/insight-folding-actions.ts:31]: Looks like this should be uncommented and used instead of the one from `../folding/folding-actions`.

File: packages/headless/src/features/insight-search/insight-search-request.ts [hunk 20]
```diff
@@ -10,7 +11,7 @@ import {MappedSearchRequest, mapSearchRequest} from '../search/search-mappings';
 
 type StateNeededBySearchRequest = ConfigurationSection &
   InsightConfigurationSection &
-  Partial<InsightAppState>;
+  Partial<InsightAppState & FoldingSection>;
```
Comment (btaillon) [inline packages/headless/src/features/insight-search/insight-search-request.ts:14]: `InsightAppState` already includes `FoldingSection`

File: packages/headless/src/controllers/core/folded-result-list/headless-core-folded-result-list.ts (hunk 15 above)
Comment (SimonMilord) [reply packages/headless/src/controllers/core/folded-result-list/headless-core-folded-result-list.ts:180]: you are right, ill remove it

File: packages/headless/src/features/folding/folding-actions.ts (hunk 18 above)
Comment (SimonMilord) [reply packages/headless/src/features/folding/folding-actions.ts:70]: loadCollection is used in both useCases, but the insight one has the InsightConfigurationSection on top of the what the folding-action. I thought I could declare it in one and import it, but following your other comment in insight-folding-actions.ts it would be better maybe to declare it in each file

File: packages/headless/src/controllers/core/folded-result-list/headless-core-folded-result-list.ts (hunk 17 above)
Comment (SimonMilord) [reply packages/headless/src/controllers/core/folded-result-list/headless-core-folded-result-list.ts:256]: for the reducers I have used the same ones previous in use by the `headless-folded-result-list` controller.
It seems to work fine without the query reducer but I believe the 3 other ones to be necessary

File: packages/headless/src/api/service/insight/insight-api-client.ts (hunk 9 above)
Comment (mmitiche) [reply packages/headless/src/api/service/insight/insight-api-client.ts]: It's more about the action that will be executed on the Search API. In the insight use case it's a bit special because the requests are not sent directly sent to the search API, they are sent to Customer Service API which performs the request to the search API. so the finality of the request is a search request.
That's why imo the value should be `search`.

**Review by mmitiche — COMMENTED** [review]
I see that the Atomic Folding  e2e test is failing, is this due to some flakiness or there is something wrong with these changes?
Folding

File: packages/headless/src/controllers/core/folded-result-list/headless-core-folded-result-list.ts (hunk 17 above)
Comment (btaillon) [reply packages/headless/src/controllers/core/folded-result-list/headless-core-folded-result-list.ts]: I'm a bit suspicious of the search reducer. Does the insight use case typically use the search reducer?

Comment (SimonMilord) [reply packages/headless/src/controllers/core/folded-result-list/headless-core-folded-result-list.ts]: I dont think the insight usecase uses it. I tested it on my end, both insight and search use cases seem to work even without the search reducer.
I would have thought it would be needed since the core controller contains both the search and insight usecases and the previous "search only" iteration of the controller seemed to add it to its engine

Comment (SimonMilord) [reply packages/headless/src/controllers/core/folded-result-list/headless-core-folded-result-list.ts]: I dug a bit more into this, turns out you were right search doesnt seem to be needed. `query`, however does.

File: packages/headless/src/controllers/core/folded-result-list/headless-core-folded-result-list.test.ts [hunk 21]
```diff
@@ -108,14 +150,14 @@ describe('FoldedResultList', () => {
...
 
-    it('dispatches analytics and folding/loadCollection when calling loadCollection', () => {
-      foldedResultList.loadCollection(foldedResultList.state.results[0]);
-      expect(engine.actions.pop()?.type).toEqual(
-        'analytics/folding/showMore/pending'
-      );
-      expect(engine.actions.pop()?.type).toEqual(
-        'folding/loadCollection/pending'
-      );
+    it('#loadCollection dispatches #loadCollection AND #logShowMoreFoldedResults analytics', () => {
```
Comment (mmitiche) [inline packages/headless/src/controllers/core/folded-result-list/headless-core-folded-result-list.test.ts:153]: #loadCollection dispatches #loadCollection sounds a bit odd, I believe we can improve the description of this test.

File: packages/headless/src/controllers/core/folded-result-list/headless-core-folded-result-list.test.ts [hunk 22]
```diff
@@ -108,14 +150,14 @@ describe('FoldedResultList', () => {
...
+      const expectedLogShowMoreAction = 'analytics/folding/showMore/pending';
+      const expectedLoadCollectionAction = 'folding/loadCollection/pending';
+
+      foldedResultList.loadCollection(foldedResultList?.state?.results[0]);
+      expect(foldedResultList.state.results).not.toBe(undefined);
```
Comment (mmitiche) [inline packages/headless/src/controllers/core/folded-result-list/headless-core-folded-result-list.test.ts:158]: why do we need this expect here? we are testing  the dispatch of `loadCollection` and `logShowMoreFoldedResults` no?

**Review by mmitiche — APPROVED** [review]
GG 👌

File: packages/headless/src/controllers/core/folded-result-list/headless-core-folded-result-list.test.ts (hunk 22 above)
Comment (SimonMilord) [reply packages/headless/src/controllers/core/folded-result-list/headless-core-folded-result-list.test.ts:160]: oh I tried to add this before I moved the test, I can remove it as I thought it could fix the test failing originally.

**Review by erocheleau — COMMENTED** [review]
Just one suggestion where I think we could simplify the code

File: packages/headless/src/controllers/core/folded-result-list/headless-core-folded-result-list.ts [hunk 23]
```diff
@@ -0,0 +1,274 @@
...
+   * The options for the `FoldedResultList` controller.
+   * */
+  options?: FoldedResultListOptions;
+  /**
+   * The action creator to build the `loadCollection` action.
+   */
+  loadCollectionActionCreator: (
+    collectionId: CollectionId
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  ) => AsyncThunkAction<any, CollectionId, any>;
```
[the remark this answers is not in the document]
Comment (SimonMilord) [reply packages/headless/src/controllers/core/folded-result-list/headless-core-folded-result-list.ts:83]: I believe so yes, other examples using the actionCreator pattern are also passing any as types in the AsyncThunkAction : 

image

File: packages/headless/src/features/folding/folding-insight-analytics-actions.test.ts [hunk 24]
```diff
@@ -0,0 +1,130 @@
...
+    );
+    expect(mockToUse.mock.calls[0][2]).toStrictEqual(expectedCaseContext);
+  });
+
+  it('should log #logShowLessFoldedResults', async () => {
+    await engine.dispatch(logShowLessFoldedResults());
+
+    const mockToUse = mockLogShowLessFoldedResults;
+    expect(mockToUse).toBeCalledTimes(1);
+    expect(mockToUse.mock.calls[0][0]).toStrictEqual(expectedCaseContext);
```
[the remark this answers is not in the document]
Comment (SimonMilord) [reply packages/headless/src/features/folding/folding-insight-analytics-actions.test.ts:128]: no we don't! :D

File: packages/headless/src/controllers/core/folded-result-list/headless-core-folded-result-list.ts [hunk 25]
```diff
@@ -0,0 +1,274 @@
...
+}
+
+/**
+ * The `CoreFoldedResultList` headless controller re-organizes results into hierarchical collections (a.k.a. threads).
+ */
+export interface CoreFoldedResultList extends Controller {
```
[the remark this answers is not in the document]
Comment (SimonMilord) [reply packages/headless/src/controllers/core/folded-result-list/headless-core-folded-result-list.ts]: good point, I will refactor these, only thing is the props which I will have to export from the files because of the actionCreator pattern. It does clean up the files nicely though thanks!

File: packages/headless/src/api/service/insight/insight-api-client.ts [hunk 26]
```diff
@@ -81,10 +82,12 @@ export class InsightAPIClient {
   }
 
   async query(
-    req: InsightQueryRequest
+    req: InsightQueryRequest,
+    options?: SearchOptions
   ): Promise<InsightAPIResponse<SearchResponseSuccess>> {
     const response = await PlatformClient.call({
       ...buildInsightQueryRequest(req),
+      requestMetadata: {method: 'search', origin: options?.origin},
```
[the remark this answers is not in the document]
Comment (SimonMilord) [reply packages/headless/src/api/service/insight/insight-api-client.ts:90]: we need this optional param to be passed in order to pass the `origin` param as it is used as a second parameter in the `search` method of the `loadCollection` action. Its use is to pass the SearchOrigin which can take these values:

`export type SearchOrigin =
  | 'mainSearch'
  | 'facetValues'
  | 'instantResults'
  | 'foldingCollection';`

File: packages/headless/src/controllers/core/folded-result-list/headless-core-folded-result-list.ts (hunk 6 above)
Comment (erocheleau) [reply packages/headless/src/controllers/core/folded-result-list/headless-core-folded-result-list.ts:72]: Yea okay if I understand what you told me correctly Simon and I don't remember why we need that in the props there, but @mmitiche is correct.

Name this one `CoreFoldedResultListProps`

File: packages/headless/src/controllers/insight/folded-result-list/headless-insight-folded-result-list.ts [hunk 27]
```diff
@@ -0,0 +1,52 @@
...
+export type {
+  FoldedResultListOptions,
+  FoldedResultListState,
+  FoldedResultList,
+  FoldingOptions,
+  FoldedCollection,
+  FoldedResult,
+};
+
+export interface FoldedResultListProps {
```
Comment (erocheleau) [inline packages/headless/src/controllers/insight/folded-result-list/headless-insight-folded-result-list.ts:23]: You should rename this something like `InsightFoldedResultListProps` to avoid confusion.

Comment (erocheleau) [inline packages/headless/src/controllers/insight/folded-result-list/headless-insight-folded-result-list.ts:23]: This should be something like : `InsightFoldedResultListProps`

**Review by erocheleau — APPROVED** [review]
Just a rename otherwise it looks good
